pkg/controllers/core/deployer_controller.go

```diff
@@ -396,11 +473,55 @@ func (r *DeployerReconciler) reconcileChildService(ctx context.Context, log logr
 	return service, nil
 }
 
+func (r *DeployerReconciler) ingressSemanticEquals(desiredIngress, ingress *v1beta1.Ingress) bool {
+	return equality.Semantic.DeepEqual(desiredIngress.Spec, ingress.Spec) &&
+		equality.Semantic.DeepEqual(desiredIngress.ObjectMeta.Labels, ingress.ObjectMeta.Labels)
+}
+
 func (r *DeployerReconciler) serviceSemanticEquals(desiredService, service *corev1.Service) bool {
 	return equality.Semantic.DeepEqual(desiredService.Spec, service.Spec) &&
 		equality.Semantic.DeepEqual(desiredService.ObjectMeta.Labels, service.ObjectMeta.Labels)
 }
 
+func (r *DeployerReconciler) constructIngressForDeployer(deployer *corev1alpha1.Deployer) (*v1beta1.Ingress, error) {
+	labels := r.constructLabelsForDeployer(deployer)
```

Comment:
Only create the ingress if the service is created. We'll also want to make ingress opt-in at some point.

```suggestion
	if deployer.Status.ServiceName == "" {
		return nil, nil
	}

	labels := r.constructLabelsForDeployer(deployer)
```